components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
+ 
+   if ( arguments.length === 1 || arguments.length === 3 ) { 
+       if ( _.isUndefined( states[inportName] ) ) { 
+           return;
+       } 
+       return states[inportName];
+   }
+
+   if ( arguments.length === 2 || arguments.length === 4 ) { 
+       states[inportName] = { data: arguments[arguments.length-1],
+                              lm: Lm(),
+                              previousLms: [],
```

Comment:
I think previousLms should be undefined if it has not yet been set -- not an empty list, because an empty list would correspond to a node with no inputs (which is possible).